lib/game.rb

```diff
@@ -0,0 +1,72 @@
+class Game
+  attr_accessor :player_1, :player_2, :board
+  WIN_COMBINATIONS = [
+    [0,1,2], # Top row
+    [3,4,5],  # Middle row
+    [6,7,8],
+
+    [0,3,6], #verticals
+    [1,4,7],
+    [2,5,8],
+
+    [0,4,8], #diagonals
+    [6,4,2]
+    ]
+
+  def initialize(player_1 = Player::Human.new("X"), player_2 = Player::Human.new("O"), board = Board.new)
+    @player_1 = player_1
+    @player_2 = player_2
+    @board = board
+  end
+
+  def current_player
+    @board.turn_count.even? ? @player_1 : @player_2
+  end
+
+  def over?
+    won? || draw?
+  end
+
+  def won?
+    WIN_COMBINATIONS.detect do |combo|
+      @board.cells[combo[0]] == @board.cells[combo[1]] && @board.cells[combo[1]] == @board.cells[combo[2]] && @board.taken?(combo[0] + 1) #@board.taken?(combo[0] + 1) to make sure it == "X" or "O" and not empty spaces, +1 for offsetting to user input
+    end
+  end
+
+  def draw?
+    @board.full? && !won?
+  end
+
+  def winner
+    !won? || draw? ? nil : @board.cells[won?.first]
```

Comment:
Avoid returning `nil` when using the ternary operator. That is usually an indication that you can simplify your expression. In this case, make your last return value the default. It could look something like this
```
if !(!won? || draw?)
  @board.cells[won?.first]
end
```
Remember that, if the `if` statement criteria is not satisfied, this expression will return nil.

Next, we can simplify a little further by distributing the negation. This might be a little confusing, but `!(a || b)` <=> `!a && !b` (this property is one of [DeMorgan's Laws](https://en.wikipedia.org/wiki/De_Morgan%27s_laws)). So, we can rewrite the updated method as
```
if won? && !draw?
  @board.cells[won?.first]
end
```
But reading that aloud, you may realize that `!draw?` is redundant; anytime `won?` returns true, `!draw?` will also return true.

That was a lot and I hope I didn't overwhelm you. Feel free to reach out to discuss more.